### PR TITLE
Add ability to set spells in Shrine in the Editor

### PR DESCRIPTION
**Pull request by ihhub**
image

image

this pull request should be merged after #9335

### CODE REVIEW

**Review by Branikolog — APPROVED** [review]

**Review by Districh-ru — COMMENTED** [review]
Hi, @ihhub, well done! 👍 
I left one optional suggestion, could you please take a look in you spare time?
Also later (in a new PR) we can optimize spells rendering not to fully redraw all spells and scrolls on the single spell change. :)

File: src/fheroes2/world/world_loadmap.cpp [hunk 1]
```diff
@@ -1012,6 +1015,71 @@ bool World::loadResurrectionMap( const std::string & filename )
                     tileData[0] = tileData[0] - 1U;
                 }
             }
+            else if ( object.group == Maps::ObjectGroup::ADVENTURE_POWER_UPS ) {
+                assert( object.index < powerUpsObjects.size() );
+
+                const auto areSpellsValid = []( const Maps::Map_Format::ShrineMetadata & metadata, const int spellLevel ) {
```
[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/world/world_loadmap.cpp:1021]: Hi @Districh-ru , the idea behind the lambda function was that it should not know the internal logic for objects. This lambda function will be used in the setting a spell for Pyramid object.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/world/world_loadmap.cpp:1021]: If we want to reduce code duplication we can use the same code as for the Editor:
```cpp

int spellLevel = 0;
if ( objectType == MP2::OBJ_SHRINE_FIRST_CIRCLE ) {
   spellLevel = 1;
}
   else if ( objectType == MP2::OBJ_SHRINE_SECOND_CIRCLE ) {
   spellLevel = 2;
}
   else if ( objectType == MP2::OBJ_SHRINE_THIRD_CIRCLE ) {
   spellLevel = 3;
}
else {
   assert( 0 );
   spellLevel = 1;
}
```

Would it be a better solution than now?

[the remark this answers is not in the document]
Comment (Districh-ru) [reply src/fheroes2/world/world_loadmap.cpp:1021]: Sure, then lets not update the lambda but IMHO it is better to combine the same code.
We can also add a function that will return `spellLevel` by `objectType` and use it in editor and here.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/world/world_loadmap.cpp:1021]: Hi @Districh-ru , I've addressed your comment. I did not create a separate function as it is going to be a very specific function just for this corner case.

**Review by oleg-derevenetz — APPROVED** [review]

**Comment by ihhub** [comment]
Hi @Districh-ru , I merged this pull request as I would like to add functionality for Pyramid before the release. Please let me know if you have any further comments so I could address them in the new pull request.

**Comment by Districh-ru** [comment]
Hi, @ihhub, I've checked the updated code and all is OK. I like that now it correlates with the editor's code so it'll be easier to maintain the same looking logic. :)

**Pull request merged**